packages/browser/src/errors.ts

```diff
@@ -0,0 +1,34 @@
+import { NormalizeKeyPaths } from '@silverhand/essentials';
+import get from 'lodash.get';
+
+const logtoClientErrorCodes = Object.freeze({
+  sign_in_session: {
+    invalid: 'Invalid sign-in session.',
+    not_found: 'Sign-in session not found.',
+  },
+});
+
+export type LogtoClientErrorCode = NormalizeKeyPaths<typeof logtoClientErrorCodes>;
+
+const getMessageByErrorCode = (errorCode: LogtoClientErrorCode): string => {
+  // TODO: linear issue LOG-1419
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const message = get(logtoClientErrorCodes, errorCode);
+
+  if (typeof message === 'string') {
+    return message;
+  }
+
+  return errorCode;
+};
+
+export class LogtoClientError extends Error {
```

Comment:
this can be extracted into a generic (error code) class? cc @simeng-li @wangsijie @demonzoo

Reply:
Added the Linear issue: [Extract generic error code class](https://linear.app/silverhand/issue/LOG-1557/extract-generic-error-code-class)